### ISSUE TITLE
Investigate MCC and ALiVE

### ISSUE THREAD

**Issue by Fourjays** [blackwatchint/blackwatchint#198]
We want to investigate integrating one or both of the MCC and ALiVE mods into our modset for mission makers.

Both mods need checking for suitability for use, performance and compatibility with one another.

**Comment by adios113** [comment]
http://www.alivemod.com/war-room/showorbat/1704

From my previous tests of ALIVE - BWI group I created back then at the beggining of 2015.

**Comment by ShiFoo** [comment]
I have just discovered the biggest downside of Alive:
It only supports certain faction and __certain maps__:

http://alivemod.com/wiki/index.php/Supported_Factions
http://alivemod.com/wiki/index.php/Supported_Maps

I tried to recreate my last mission on Bozcaada. It crashed with an error.
For the maps, it would be needed to index them outselves which seems to be an immense effort.

**Comment by ShiFoo** [comment]
I have now also tested MCC. We can not avoid placing MCC modules in the templates to preset some configuration.

__Pros:__
- Once understood, zones are pretty straight forward and easy to use.
- GAIA will provide more of a challenge than regular Zeus AI.
- Caching of AI units.
- Can set detailed skill level of AI (Commanding skill, Aiming skill, spotting skill, ...)
- AI stances (aggressive, defensive, garrison, patrol).
- Paradrop out of chopper with a click of a button. 
- HALO jumping

__Cons:__
- No way of regularing the density of AI when using the Spawn AI function. It will populate the zone relatively dense.
- Impossible to delete cached AI when hidden (player not near them). 
- Has a lot of "arcadey" functions that we will most likely not/should not use.
- Looks like once an AI unit is spawned through MCC on the server, you can't reassign it to the HC afterwards.

We always had full control over AI (placing and rotating every single unit). With the "spawn in the zones", the Zeus needs to understand that he has to focus on more general things now and let GAIA do the work. Also, the withdawal of control will be uncomfortable in the beginning.

Even though i listed some cons, i think the pros outweighs them. I will continue my testing but so far MCC looks promising.

**Comment by Fourjays** [comment]
Regards ALiVE, I knew factions had to be configured but not maps. That's a bit of a limitation, although not one that we couldn't workaround. Do we know if units also need submitting to the ALiVE team?

**Comment by abelarno** [comment]
I have experience with ALiVE from a previous unit. Here are some quick pro's and con's

**Pro's**
* Very customizable. Almost all settings can be changed trough editor modules.
* Replayability. You can use the same mission for different operations. Especially on larger maps. ALiVE randomizes everything within the set parameters on the first mission load, making almost everything different.
* Persistent. At the end of an operation, the server admin can sync the mission to the alive servers. The next time that mission is loaded, everything will be where it was (can be turned off for players).
* Immersive. Battles can happen in town with civilians. Shooting them will harm your relation with the local population, and might even turn them against you to the point where they will pick up arms. Liberating them on the other hand might make them friendly to you, and they might give critical intel. You have to talk to the population to find this out.

**Con's**
* Limited control over the mission. This means that an active zeus might need to intervene to make sure people don't die of boredom or too many enemies.
* Long loading times. loading the map on the server can take ~10 minutes, depending on the amount of forces present.
* Limited amount of maps and units. RHS works tough.
* Requires set-up in the editor, and might be hard to balance properly.


In short I think that ALiVE would be best in bigger campaigns, that would span over a bunch of nights. It does have it's place on one off operations, but then in a more supportive role, and we should not be depending on it then.

**Comment by Fourjays** [comment]
Is there any way to disable MCC for specific templates? In my tests MCC ran by default.

**Comment by abelarno** [comment]
You can place settings modules down that can turn it all off. I don't believe there is a "master switch" tough.

**Comment by ShiFoo** [comment]
The templates needs the modules of MCC. You can define basic MCC config parameters there. Most of the functionality is disabled anyways, except we agree on enabling it. And as far as i know, as long as you dont log into the MCC console and start spawning units, it should have no effect on the mission itself. The option to launch MCC console is also defined in the templates where the zeus placers are synched with the Access Right module.

**Comment by ShiFoo** [comment]
Another point would be that ALiVE has no equivalent of GAIA. But in that regard, maybe ASR can prove helpful as Abel was using both in parallel in his previous unit.

**Comment by abelarno** [comment]
ASR has no modules to turn it on or off, or any kind of settings. The mod is either loaded and on, or not loaded.

**Comment by ShiFoo** [comment]
I have seen that you can trigger ASR in the configfile in userconfig\asr_ai3\asr_ai3_settings_ss.sqf
But that doesnt really help us. What i could try is to enhance his mod with a editor placeable module where you can set the "enable" to yes or no.

**Comment by abelarno** [comment]
Found out that ASR allows some settings to be changed in the init.sqf file. That includes an on/off  option. No idea how I missed that before.
This makes it a viable option over MCC.

**Comment by ShiFoo** [comment]
asr_ai3_main_enabled = 0; will disable it. It just gives us the most flexible solution and we can pretty much have any combination now.

**Comment by Fourjays** [comment]
Server-side ALiVE testing is being actively performed (#232) in preparation for the inclusion of both ALiVE and ASR AI (#247 and #248).

Closed as research stage is complete.
